homeassistant/components/shelly/climate.py

```diff
@@ -308,6 +318,9 @@ async def async_added_to_hass(self) -> None:
         LOGGER.info("Restoring entity %s", self.name)
 
         last_state = await self.async_get_last_state()
+        last_extra_data = await self.async_get_last_extra_data()
+        if last_extra_data is not None:
+            self._last_target_temp = last_extra_data.as_dict()["last_target_temp"]
```

Comment:
```suggestion
            self._last_target_temp = last_extra_data.last_target_temp
```

Reply:
With this change I think we can also drop the `as_dict` method

Reply:
I always get `AttributeError: 'RestoredExtraData' object has no attribute 'last_target_temp'` when I access `last_extra_data.last_target_temp`. Also with casting `cast(ShellyClimateExtraStoredData, last_extra_data).last_target_temp`.

We can access the value by `last_extra_data.json_dict["last_target_temp"]`, I think it looks better.

Reply:
I think it is acceptable to leave it as it is now `self._last_target_temp = last_extra_data.as_dict()["last_target_temp"]` until we find a better option or improve the `async_get_last_extra_data` helper. I would not change it to `last_extra_data.json_dict["last_target_temp"]`

Reply:
We should leave it as-is. We are stuck with the current restore design since it can't restore the type/class given it gets serialized to JSON.  We could convert it back into `ShellyClimateExtraStoredData` but since we only use it here its likely not worth it.